backend/src/main/java/com/woowacourse/levellog/team/domain/Team.java

```diff
@@ -22,168 +22,94 @@
     }
 
-    private void validateTitle(final String title) {
-        if (title == null || title.isBlank()) {
-            throw new InvalidFieldException("팀 이름이 null 또는 공백입니다.", DebugMessage.init()
-                    .append("title", title));
-        }
-        if (title.length() > DEFAULT_STRING_SIZE) {
-            throw new InvalidFieldException("팀 이름은 " + DEFAULT_STRING_SIZE + " 이하여야 합니다.", DebugMessage.init()
-                    .append("title 길이", title.length()));
-        }
+    public void delete(final LocalDateTime presentTime) {
+        validateReady(presentTime);
+
+        participants.clear();
+        deleted = true;
     }
 
-    private void validatePlace(final String place) {
-        if (place == null || place.isBlank()) {
-            throw new InvalidFieldException("장소가 null 또는 공백입니다.", DebugMessage.init()
-                    .append("place", place));
-        }
-        if (place.length() > DEFAULT_STRING_SIZE) {
-            throw new InvalidFieldException("장소 이름은 " + DEFAULT_STRING_SIZE + " 이하여야 합니다.", DebugMessage.init()
-                    .append("place 길이", place.length()));
-        }
+    public void close(final LocalDateTime presentTime) {
+        validateInProgress(presentTime);
+        isClosed = true;
     }
 
-    private void validateStartAt(final LocalDateTime startAt) {
-        if (startAt == null) {
-            throw new InvalidFieldException("시작 시간이 없습니다.", DebugMessage.init()
-                    .append("startAt", startAt));
-        }
+    public TeamStatus status(final LocalDateTime presentTime) {
+        return TeamStatus.of(isClosed, detail.getStartAt(), presentTime);
+    }
 
-        final LocalDateTime now = LocalDateTime.now();
-        if (now.isAfter(startAt)) {
-            throw new InvalidFieldException("인터뷰 시작 시간은 현재 시간 이후여야 합니다.", DebugMessage.init()
-                    .append("startAt", startAt)
-                    .append("now", now));
+    public void validateReady(final LocalDateTime presentTime) {
+        if (status(presentTime) != TeamStatus.READY) {
+            throw new TeamNotReadyException(DebugMessage.init()
+                    .append("presentTime", presentTime));
         }
     }
 
-    private void validateProfileUrl(final String profileUrl) {
-        if (profileUrl == null || profileUrl.isBlank()) {
-            throw new InvalidFieldException("팀 프로필 사진이 null 또는 공백입니다.", DebugMessage.init()
-                    .append("profileUrl", profileUrl));
-        }
-        if (profileUrl.length() > PROFILE_URL_SIZE) {
-            throw new InvalidFieldException("잘못된 팀 프로필 사진을 입력했습니다.", DebugMessage.init()
-                    .append("profileUrl", profileUrl));
+    public void validateInProgress(final LocalDateTime presentTime) {
+        final TeamStatus status = status(presentTime);
+        if (status == TeamStatus.CLOSED) {
+            throw new TeamAlreadyClosedException(DebugMessage.init()
+                    .append("teamId", getId()));
```

Comment:
```suggestion
        status(presentTime)
            .validateInProgress(getId(), presentTime));
```
이런식으로 `TeamStatus`에게 책임을 위임할 수 있겠네요.

Reply:
고민점
- 로그 메시지를 남기기 위해 TeamStatus 검증 메소드에서 teamId 또는 presentTime을 매개변수로 받아야 함

Reply:
`TeamStatus` 에 해당 로직 외에 Ready 검증 로직도 넣어주었습니다. TeamStatus가 관리할 책임이 맞다고 생각이 드네요.